src/main/java/net/sf/jabref/logic/citationstyle/CitationStyleCache.java

```diff
@@ -0,0 +1,86 @@
+package net.sf.jabref.logic.citationstyle;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.sf.jabref.BibDatabaseContext;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.event.EntryChangedEvent;
+import net.sf.jabref.model.event.EntryRemovedEvent;
+
+import com.google.common.eventbus.Subscribe;
+
+
+/**
+ * Caches the generated Citations for quicker access
+ */
+public class CitationStyleCache {
+
+    private CitationStyle citationStyle;
+    private Map<String, Map<BibEntry, String>> citationStylesCache = new HashMap<>();
```

Comment:
I would only cache for one citation style and reset the whole cache if another style is used (in `setCitationStyle`). That makes the code more understandable and I suspect most people actually use just one citation style or don't constantly switch between them (at least I never used the second preview). 
Or are there good reasons to store them for different styles?